### PR TITLE
Product uploading workflow

### PR DESCRIPTION
**Pull request by ilci**
Add further infos to product_uploader.info.yml and write the product_uploader.module. Activate the module on the site.

### CODE REVIEW

File: web/modules/custom/product_uploader/product_uploader.module [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+declare(strict_types = 1);
+
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ */
+function product_uploader_user_presave(EntityInterface $entity): void {
+  $fragment = '+productuploader';
```
Comment (tamasd) [inline web/modules/custom/product_uploader/product_uploader.module]: Move this to a constant outside of this function using the `const` keyword. Prefix the constant's name with the module's name. It should be something like `PRODUCT_UPLOADER_FRAGMENT`.

File: web/modules/custom/product_uploader/product_uploader.module [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  $email = $entity->getEmail();
+  if (strpos($email, $fragment)) {
+    $entity->addRole('product_uploader');
```
Comment (tamasd) [inline web/modules/custom/product_uploader/product_uploader.module]: Move the name of the role into a constant.

File: web/modules/custom/product_uploader/product_uploader.module [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  }
+  else {
+    $entity->removeRole('product_uploader');
```
Comment (tamasd) [inline web/modules/custom/product_uploader/product_uploader.module]: Move the name of the role into a constant.

File: web/modules/custom/product_uploader/product_uploader.module [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  }
+}
+
+/*https://red-route.org/articles/adding-roles-user-programmatically-drupal-8
```
Comment (tamasd) [inline web/modules/custom/product_uploader/product_uploader.module]: Remove comment.

**Review by tamasd — CHANGES_REQUESTED** [review]

File: web/modules/custom/product_uploader/src/ProductUploaderInterface.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\product_uploader;
+
+/**
+ * Class for the Product Uploder role.
+ */
+class ProductUploaderInterface {
```
Comment (tamasd) [inline web/modules/custom/product_uploader/src/ProductUploaderInterface.php]: Is this a class or an interface?

I think this should be a `final class` with an empty private constructor: https://stackoverflow.com/a/53194084

**Review by tamasd — CHANGES_REQUESTED** [review]

**Review by tamasd — APPROVED** [review]

**Review by kittiradovics — APPROVED** [review]

**Pull request merged**